js/test/setupTest.ts

```diff
@@ -1,5 +1,11 @@
 import "@testing-library/jest-dom";
+import { neverPromise } from "./helpers/promiseWithResolvers";
 // add all jest-extended matchers
 import * as jestExtendedMatchers from "jest-extended";
 
 expect.extend(jestExtendedMatchers);
+
+jest.mock("../browser", () => ({
+  fetch: jest.fn(() => neverPromise()),
```

Comment:
Could a non-browser runtime appear to hang if it refuses to exit until this never-fulfilling promise finishes? Or rely on the testing framework's timeout?

IDK, just seems like an anti-pattern.

Reply:
I don't think that other runtimes like the one in which our Jest tests run would generally block on a promise until it resolves / rejects, otherwise you couldn't interleave other actions between the creation of the promise and its fulfillment / rejection.

Reply:
This promise won't control when the test exits (unless you put `await fetch()` in your test). It won't cause hitting the jest timeout (and I just verified this by playing with the timeout setting).

This will mean that any components that make api calls (unless they mock fetch separately) will show loading throughout the test, which seemed the most stable and predictable behavior. Other options are:
- not giving an implementation (it returns undefined, and `fetch().then()` crashes)
- always resolving (no good choice for what to return)
- always rejecting (I think always showing loading is a better default behavior for a test than always showing an error, given all the ways we might respond to an error)

Any test where the result of a fetch matters (or where it matters whether it resolves) should pass their own mock implementation that returns a promise that the test controls.

Reply:
For instance, in [this test](https://github.com/mbta/orbit/blob/main/js/test/hooks/useNfc.test.ts#L23) the promise does eventually get rejected, but other things happen between the creation of the promise and the call to `reject`, and those aren't blocked.

Reply:
OK! My concern is assuaged 👍